### DISCUSSION TITLE
i can't use firebase dataprovider

### DISCUSSION THREAD

**Discussion by pinale** [refinedev/refine#2999] [Q&A] [answered]
i know firebase dataprovider is a third pary dataprovider and is not part of refine project, but is someone of you  able to use it successfully?
i've tried to integrateit  but i've some types mistmatch by typescript level

I've open an issue in the owner repo but the project doesn't seem mantaine anymore 

this is my repo.
https://github.com/pinale/Refine/tree/master/refinefirebase

Can someone help me please?

**Comment by aliemir** [comment]
Hey @pinale thank you for reaching out. Firebase data provider is not maintained by our team but we've subscribed to the issue on @rturan29's repo and watching it https://github.com/rturan29/refine-firebase/issues/5#issuecomment-1314837975.

**Reply by pinale** [reply]
ok thanks, i've added more details on that issue!

**Reply by resulturan** [reply]
I will check the issue.

**Reply by resulturan** [reply]
I checked the issue and the problem may be about the type definition of dataProvider props of Refine.

default field is required in IDataMultipleContextProvider but missing in IDataContextProvider.

**Reply by pinale** [reply]
So this `missing field` is something that it is missing in the definition of these interfaces? and not something that **I** should have to pass.
Correct?

**Comment by pinale** [comment]
@aliemir can you tell me something?

**Comment by aliemir** [comment, marked as answer]
Hey @pinale and @rturan29. Sorry for my late response. 

@pinale There were some changes made to the types of data providers and `refine-firebase` does not use `@pankod/refine-core` to pick and use types; instead it uses a copy 🤔 (Trying to explain this a bit more below)

@rturan29 Thank you for the help, looks like in `refine-firebase` you didn't use types from `@pankod/refine-core` to keep them in sync 🤔 To update the types to latest and fix them (if necessary); we can do like;

- Add the `@pankod/refine-core` to `peerDependencies`. (with `^` of course) 
- Add the `@pankod/refine-core` to `devDependencies` if you want to include this in the tests.
- Replace the types from `@pankod/refine-core` to get them in sync; (not-exported types can be extracted from the components) 
- If there are changes needed to be made because of the type changes, I'll try to help on adjusting them. 🤝 

- There's a brand new data provider, just released from our community 🎉  (Thank you @chirdeeptomar 🙏 ) [`refine-elide-rest`](https://github.com/chirdeeptomar/refine-elide-rest). You can check it to adjust types https://github.com/chirdeeptomar/refine-elide-rest/blob/main/src/index.ts 🤔 I think it can help 

@pinale If you can check these changes, I think you can help @rturan29 with updating the package to the latest 🚀 🚀 How's that sound @rturan29 and @pinale ?

**Reply by pinale** [reply]
i made this PR, i hope it can be good enough e to be merged
https://github.com/resulturan/refine-firebase/pull/6
feedback are appreciated
